src/main/java/org/mockito/internal/framework/DefaultMockitoFramework.java

```diff
@@ -0,0 +1,32 @@
+package org.mockito.internal.framework;
+
+import org.mockito.MockitoFramework;
+import org.mockito.internal.progress.MockingProgress;
+import org.mockito.listeners.MockCreationListener;
+import org.mockito.listeners.MockitoListener;
+import org.mockito.listeners.StubbingListener;
+
+import static org.mockito.internal.progress.ThreadSafeMockingProgress.mockingProgress;
+
+public class DefaultMockitoFramework implements MockitoFramework {
+
+    //TODO SF!
+    public void setStubbingListener(StubbingListener listener) {
+        MockingProgress p = mockingProgress();
+        if (listener == null) {
+            p.setStubbingListener(null);
+        } else {
+            p.setStubbingListener(new ThreadSafeStubbingListener(listener));
+        }
+    }
+
+    public void addListener(MockitoListener listener) {
+        if (listener instanceof MockCreationListener) {
```

Comment:
@szczepiq 

Is this really needed? [MockingProgressImpl.mockingStarted()] (https://github.com/mockito/mockito/pull/527/files#diff-df53f45103af61fe77db9216591b7ad5R146) filters already for `MockCreationListener`.